rust/osm2lanes/src/tag/osm.rs

```diff
@@ -0,0 +1,136 @@
+use crate::tag::{TagKey, Tags};
+
+pub const HIGHWAY: TagKey = TagKey::from("highway");
+
+pub enum Highway {
+    Road(Road),
+    Link(Road),
+    Bridleway,
+    BusGuideway,
+    Construction,
+    Corridor,
+    Cycleway,
+    Escape,
+    Footway,
+    LivingStreet,
+    Path,
+    Pedestrian,
+    Proposed,
+    Raceway,
+    Residential,
```

Comment:
Why isn't residential considered a road? Or service roads? (What's the difference between Road and Highway -- is it just stuff that can have a `_link` suffix?)